### PR TITLE
Tagging the image with the specific branch

### PR DESCRIPTION
**Pull request by mitangelo**
Adding the ability to build a specific branch of openedx. 
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Enhancement (improves on existing behavior)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project. (Did you install and run the pre-commit hooks?)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by mitangelo** [comment]
Will be adding the remaining config files.

**Comment by blarghmatey** [comment]
Looks good, though the pipeline still needs to switch from the `github-release` to the `git` resource for triggering builds on upstream changes

File: src/ol_infrastructure/applications/forum/Pulumi.applications.forum.mitx.Production.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+secretsprovider: awskms://alias/infrastructure-secrets-production
+encryptedkey: AQICAHg/+QzF9hGIaoayDitgnEVHEhuaANONVQQOnqpkIsol1gHaXdOgJQNVZipYBEMrjnq2AAAAfjB8BgkqhkiG9w0BBwagbzBtAgEAMGgGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMveJ6nv2z4uQIWZBVAgEQgDuZfkB3otCyaR5Z5Lz1KbFObdVmXafnFEJ2Fn91n/bGIDMrIk6INxAS5tg05t30zyu5YfnroXzP1Ib3Uw==
+config:
+  forum:auto_scale:
+    desired: 1
```
Comment (blarghmatey) [inline src/ol_infrastructure/applications/forum/Pulumi.applications.forum.mitx.Production.yaml:6]: For production we'll probably want to start with 3 instances

File: src/ol_infrastructure/applications/forum/Pulumi.applications.forum.mitx.Production.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    max: 2
+    min: 1
+  forum:business_unit: residential
+  forum:target_vpc: data_vpc
```
Comment (blarghmatey) [inline src/ol_infrastructure/applications/forum/Pulumi.applications.forum.mitx.Production.yaml:10]: This should be `residential_mitx_vpc`

File: src/ol_infrastructure/applications/forum/Pulumi.applications.forum.mitxonline.CI.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+encryptedkey: AQICAHjs8ajWpT7YRhWXwI//wPkHX53RHlo0DjkgQOwCBTUBwQE3kd5GmEzmPSXnPM2AMo5KAAAAfjB8BgkqhkiG9w0BBwagbzBtAgEAMGgGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMsLSaOWVzpdmBldJjAgEQgDuMVa6NlHhRndq9maZhpe8FLCiL3r3rYdcSP1XCBA4GcntiRGYBDmONhRMndcnhI+XztJud1gsu0jPE/Q==
+config:
+  aws:region: us-east-1
+  consul:address: https://consul-mitx-ci.odl.mit.edu
+  forum:auto_scale:
+    desired: 1
+    max: 2
+    min: 1
+  forum:business_unit: mitxonline
+  forum:target_vpc: residential_mitx_vpc
```
Comment (blarghmatey) [inline src/ol_infrastructure/applications/forum/Pulumi.applications.forum.mitxonline.CI.yaml:12]: This should be `mitxonline_vpc`

File: src/ol_infrastructure/applications/forum/Pulumi.applications.forum.mitxonline.Production.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+encryptedkey: AQICAHg/+QzF9hGIaoayDitgnEVHEhuaANONVQQOnqpkIsol1gGBr+9wC0RNsmO2noJTdlAKAAAAfjB8BgkqhkiG9w0BBwagbzBtAgEAMGgGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQM21BXoZIb5hh7mVp5AgEQgDv1ca9zhVykIhELlmD6D+gUVkevz/1iWsxJnkJOJxVAUhviUhLwLKkV8qu3xPBrZ8sTSWB88CFHkxQZcw==
+config:
+  aws:region: us-east-1
+  consul:address: https://consul-mitxonline-production.odl.mit.edu
+  forum:auto_scale:
+    desired: 1
+    max: 2
+    min: 1
+  forum:business_unit: mitxonline
+  forum:target_vpc: residential_mitx_vpc
```
Comment (blarghmatey) [inline src/ol_infrastructure/applications/forum/Pulumi.applications.forum.mitxonline.Production.yaml:12]: Here as well `mitxonline_vpc`

File: src/ol_infrastructure/applications/forum/Pulumi.applications.forum.mitxonline.Production.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+---
+secretsprovider: awskms://alias/infrastructure-secrets-production
+encryptedkey: AQICAHg/+QzF9hGIaoayDitgnEVHEhuaANONVQQOnqpkIsol1gGBr+9wC0RNsmO2noJTdlAKAAAAfjB8BgkqhkiG9w0BBwagbzBtAgEAMGgGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQM21BXoZIb5hh7mVp5AgEQgDv1ca9zhVykIhELlmD6D+gUVkevz/1iWsxJnkJOJxVAUhviUhLwLKkV8qu3xPBrZ8sTSWB88CFHkxQZcw==
+config:
+  aws:region: us-east-1
+  consul:address: https://consul-mitxonline-production.odl.mit.edu
+  forum:auto_scale:
+    desired: 1
```
Comment (blarghmatey) [inline src/ol_infrastructure/applications/forum/Pulumi.applications.forum.mitxonline.Production.yaml:8]: We probably want at least 3 instances

File: src/ol_infrastructure/applications/forum/Pulumi.applications.forum.mitxonline.QA.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+encryptedkey: AQICAHi7xhTkB8tf1ObyPMxDhODJja4Mn4jyIo32zZVlOiZPFgH/8ypsAjbdKKfa73ztbHePAAAAfjB8BgkqhkiG9w0BBwagbzBtAgEAMGgGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMM7hMwQGsReFklFDKAgEQgDvOM12LjwSWe4Rx36JFztJowDIc3yQADTSF3Lg7hPp53KrdXDQmxznqUUIK32JMfH5Cr0o9NgOvRJmH4A==
+config:
+  aws:region: us-east-1
+  consul:address: https://consul-mitxonline-qa.odl.mit.edu
+  forum:auto_scale:
+    desired: 1
+    max: 2
+    min: 1
+  forum:business_unit: mitxonline
+  forum:target_vpc: residential_mitx_vpc
```
Comment (blarghmatey) [inline src/ol_infrastructure/applications/forum/Pulumi.applications.forum.mitxonline.QA.yaml:12]: `mitxonline_vpc`

File: src/ol_infrastructure/applications/forum/Pulumi.applications.forum.xpro.CI.yaml [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+encryptedkey: AQICAHjs8ajWpT7YRhWXwI//wPkHX53RHlo0DjkgQOwCBTUBwQF08z9c11tfqNXRXfHgXiOeAAAAfjB8BgkqhkiG9w0BBwagbzBtAgEAMGgGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMEcgOiPv2JnYRU7+8AgEQgDtORMJTpQeaZFwklsSJOgv3l5zmYUqkQjOqQ3Lg+SCxAob2NzmTROGdr+BRdS/Y1974cAAq5QHdpJVYbQ==
+config:
+  aws:region: us-east-1
+  consul:address: https://consul-xpro-ci.odl.mit.edu
+  forum:auto_scale:
+    desired: 1
+    max: 2
+    min: 1
+  forum:business_unit: mitxpro
+  forum:target_vpc: residential_mitx_vpc
```
Comment (blarghmatey) [inline src/ol_infrastructure/applications/forum/Pulumi.applications.forum.xpro.CI.yaml:12]: `xpro_vpc`

File: src/ol_infrastructure/applications/forum/Pulumi.applications.forum.xpro.Production.yaml [hunk 8]
```diff
@@ -0,0 +1,16 @@
+---
+secretsprovider: awskms://alias/infrastructure-secrets-production
+encryptedkey: AQICAHg/+QzF9hGIaoayDitgnEVHEhuaANONVQQOnqpkIsol1gH/8B9HwincPwBG2gtLZZyLAAAAfjB8BgkqhkiG9w0BBwagbzBtAgEAMGgGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMTU3dSX79VQQGTRu4AgEQgDsM0riTMyEuRQX2E/Q+TMBn3AdSvo8gR5aGeebKq0gTjYHI2vfwjuVETzxFOhXMUwsh8gZ9wzGMX2vdKw==
+config:
+  aws:region: us-east-1
+  consul:address: https://consul-xpro-production.odl.mit.edu
+  forum:auto_scale:
+    desired: 1
```
Comment (blarghmatey) [inline src/ol_infrastructure/applications/forum/Pulumi.applications.forum.xpro.Production.yaml:8]: 3 instances

File: src/ol_infrastructure/applications/forum/Pulumi.applications.forum.xpro.Production.yaml [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    max: 2
+    min: 1
+  forum:business_unit: mitxpro
+  forum:target_vpc: residential_mitx_vpc
```
Comment (blarghmatey) [inline src/ol_infrastructure/applications/forum/Pulumi.applications.forum.xpro.Production.yaml:12]: `xpro_vpc`

File: src/ol_infrastructure/applications/forum/Pulumi.applications.forum.xpro.QA.yaml [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+encryptedkey: AQICAHi7xhTkB8tf1ObyPMxDhODJja4Mn4jyIo32zZVlOiZPFgEarYqGtTm7LIFq0G4ASqHvAAAAfjB8BgkqhkiG9w0BBwagbzBtAgEAMGgGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMWXlNMRJlUc070NX4AgEQgDucVokXd4ZAHaS37PgYb/MKw8blv8GXTO3O3MadAS39ZVjPyEeQ04RgJ8wpfqHVlPeRuMcIlFrqyUz9Mg==
+config:
+  aws:region: us-east-1
+  consul:address: https://consul-xpro-qa.odl.mit.edu
+  forum:auto_scale:
+    desired: 1
+    max: 2
+    min: 1
+  forum:business_unit: mitxpro
+  forum:target_vpc: residential_mitx_vpc
```
Comment (blarghmatey) [inline src/ol_infrastructure/applications/forum/Pulumi.applications.forum.xpro.QA.yaml:12]: `xpro_vpc`

**Comment by blarghmatey** [comment]
:+1: looks good!

**Pull request merged**